src/cq_cam/common.py

```diff
@@ -1,12 +1,270 @@
-from enum import Enum
+from enum import Enum, auto
+
+"""
+G-code (also RS-274) is the most widely-used computer numerical control (CNC) programming language. 
+It is used mainly in computer-aided manufacturing to control automated machine tools, as well as from a 3D-printing slicer app.
+Here we concentrate on a subset of G-code relevant for 3-axis CNC machining.
+
+The G-code syntax includes single letter addresses to configure the machine state and control the motors. 
+The two primary letter addresses are M-codes and G-codes. M-codes are known as machine codes (or more accurately miscalleneous codes) and G-codes are called preparatory codes.
+M-codes allow for state changes of the machine components and the running program. G-codes control the motion of the motors and the internal configuration of the machine. G-code commands can be catergorised as modal or non-modal.
+Modal commands remain in effect until they are replaced or cancelled by another command. Non-modal commands execute only once.
+M-code and G-code are further organised into modal groups.
+
+G-code Modal Groups
+- Group 0 - Non-modal codes: G4, G10 G28, G30, G52, G53, G92, G92.1, G92.2, G92.3 
+- Group 1 - Motion: G0, G1, G2, G3, G33, G38.n, G73, G76, G80, G81, G82, G83, G84, G85, G86, G87, G88, G89
+- Group 2 - Plane: G17, G18, G19, G17.1, G18.1, G19.1
+- Group 3 - Distance Mode: G90, G91
+- Group 4 - Arc Distance Mode: G90.1, G91.1
+- Group 5 - Feed Rate Mode: G93, G94, G95
+- Group 6 - Units: G20, G21
+- Group 7 - Cutter Diameter Compensation: G40, G41, G42, G41.1, G42.1
+- Group 8 - Tool Length Offset: G43, G43.1, G49
+- Group 10 - Canned Cycle Return Mode: G98, G99
+- Group 12 - Coordinate System: G54, G55, G56, G57, G58, G59, G59.1, G59.2, G59.3
+- Group 13 - Planner Control Mode: G61, G61.1, G64
+- Group 14 - Spindle Speed Mode: G96, G97
+
+M-code Modal Groups:
+- Group 4 - Stopping: M0, M1, M2, M30, M60
+- Group 7 - Spindle: M3, M4, M5
+- Group 8 - Coolant: M7, M8, M9
+
+In addition to the G and M address there are other letter addresses that are used in conjuction with them:
+- F - Feed Rate: G1, G2, G3
+- X - X Axis: G1, G2, G3
+- Y - Y Axis: G1, G2, G3
+- Z - Y Axis: G1, G2, G3
+- I - Arc Center in X Axis: G2, G3
+- J - Arc Center in Y Axis: G2, G3
+- K - Arc Center in Z Axis: G2, G3
+- R - Canned Cycle Arc Radius: G80, G81, G82, G83, G84, G85, G86, G88, G89
+- P - Canned Cycle Dwell Time: G80, G81, G82, G83, G84, G85, G86, G88, G89
+- L - Canned Cycle Loop Count: G80, G81, G82, G83, G84, G85, G86, G88, G89
+- S - Spindle: M3, M4
+- T - Tool Selection: M6
+- H - Tool Length Offset: G43
+- D - Radius Offset: G41, G42
+"""
+
+"""
```

Comment:
These strings can be joined together, unless there was some special purpose for splitting them.